osh/cmd_eval.py

```diff
@@ -1481,12 +1481,14 @@ def _DoRedirect(self, node, cmd_st):
         if status == 0:
             self.shell_ex.PushRedirects(redirects, io_errors)
             if len(io_errors):
-                # core/process.py prints cryptic errors, so we repeat them
-                # here.  e.g. Bad File Descriptor
-                self.errfmt.PrintMessage(
-                    'I/O error applying redirect: %s' %
-                    pyutil.strerror(io_errors[0]),
-                    self.mem.GetFallbackLocation())
+                # if errno is 0, error has already printed
+                if io_errors[0].errno:
+                    # core/process.py prints cryptic errors, so we repeat them
+                    # here.  e.g. Bad File Descriptor
+                    self.errfmt.PrintMessage(
+                        'I/O redirect error: %s' %
```

Comment:
I think the old string was more descriptive: "I/O error applying redirect"

The new one doesn't seem grammatically correct to me